### PR TITLE
[FIX] allow user to turn off mass trace length check

### PR DESCRIPTION
**Pull request by hroest**
The check for the length of the masstrace does not make sense if the
mass trace detection is followed by elution peak detection as in the
MassTraceExtractor where long mass traces are split up. Instead what
happens is that MTD is extremely slow since it finds the same masstraces
again and again but is prevented from adding them to the result and move
on.

This commit allows to set max_trace_length to -1 and skip the length
check.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
+1

**Comment by hroest** [comment]
@timosachsenberg I also wonder whether we should make -1 the default here? I think it makes way more sense especially as the EPD is enabled both for FFMetabo and MTExtractor by default. I dont see the point of having this parameter in there if we run EPD afterwards (except slowing things down to a crippling speed - the whole run literally speeds up 10x or more when I use a larger value for this parameter)

**Comment by Metabo** [comment]
In the very first version, there was no trace length check anyway. This was added later upon request (for filtering out ultra-long (most likely) contaminant traces). However, it was never clear how well it is harmonizing with the subsequent EPD.

TL;DR: I'm fine with a switch to turn off this length check.
